avocado/plugins/teststmpdir.py

```diff
@@ -0,0 +1,26 @@
+import os
+import shutil
+import tempfile
+
+from avocado.core.plugin_interfaces import JobPre, JobPost
+
+
+class TestsTmpDir(JobPre, JobPost):
+
+    name = 'teststmpdir'
+    description = 'Creates a temporary directory for tests consumption'
+
+    def __init__(self):
+        self._varname = 'XXX_TESTS_COMMON_TMPDIR'
+        self._dirname = None
+
+    def pre(self, job):
+        if os.environ.get(self._varname) is None:
```

Comment:
Cool, this is even better than what I had in mind. What this does is it allows one to share `TESTS_TMPDIR` between multiple jobs and it's only cleared when created. 👍